org.eclipse.winery.topologymodeler.ui/src/app/node/node.component.ts

```diff
@@ -143,44 +142,44 @@ export class NodeComponent implements OnInit, AfterViewInit, OnDestroy, DoCheck
      * We have 3 possibilities: none, XML element, or Key value pairs.
      * @param {string} type
      */
-    findOutPropertyDefinitionTypeForProperties(type: string, groupedNodeTypes: any): string {
-        for (const nameSpace of groupedNodeTypes) {
-            for (const nodeTypeVar of nameSpace.children) {
+    findOutPropertyDefinitionTypeForProperties(type: string, groupedNodeTypes: any): void {
+        groupedNodeTypes.some(nameSpace => {
+            nameSpace.children.some(nodeTypeVar => {
                 if (nodeTypeVar.id === type) {
                     // if PropertiesDefinition doesn't exist then it must be of type NONE
                     if (isNullOrUndefined(nodeTypeVar.full.serviceTemplateOrNodeTypeOrNodeTypeImplementation[0].propertiesDefinition)) {
-                        return 'NONE';
+                        this.propertyDefinitionType = 'NONE';
```

Comment:
Could you use a string enum for this?
See https://blog.mariusschulz.com/2017/10/27/typescript-2-4-string-enums for a how to.